src/select-text.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {onDocumentReady} from './document-ready';
+import {viewerFor} from './viewer';
+
+export function maybeSelectText(win) {
+  const text = viewerFor(win).getParam('select')
+  if (!text) {
+    return;
+  }
+  onDocumentReady(win, () => {
+    selectText(win.document.documentElement, text);
+  });
+}
+
+
+export function findNodeWithText(rootNode, string) {
+  const search = document.evaluate(
+      '//text()[contains(., "' + string + '")]',
```

Comment:
if i remember correctly, unless you add `.` to make it `.//text()...` you are actually starting from the document root (`hyml` instead of the given `rootNode` argument)